docs/proposals/2170-kubeflow-training-v2/README.md

```diff
@@ -0,0 +1,1438 @@
+                    claimName: model-initializer
+                - name: model-exporter
+                  persistentVolumeClaim:
+                    claimName: model-exporter
+    - name: Exporter
+      template:
+        spec:
+          template:
+            spec:
+              containers:
+                - name: model-exporter
+                  image: docker.io/kubeflow/model-exporter
+                  volumeMounts:
+                    - mountPath: /workspace/adapters
+                      name: model-exporter
+              volumes:
+                - name: model-exporter
+                  persistentVolumeClaim:
+                    claimName: model-exporter
+```
+
+### The Pod Spec Overrides APIs
+
+The `PodSpecOverrides` represents overrides for the `TrainingRuntime` when `TrainJob` is created.
+These parameters can include the user's identity or PVC.
+
+Usually, these parameters should not be configured by the user and should be attached during the
+orchestration (e.g. using Kubernetes admission webhooks or custom clients).
+
+In the future, we can add more parameters if we find use-cases when it is required.
+
+```golang
+type PodSpecOverride struct {
+    // Name of the training replica in the training runtime template to override
+    TargetReplicatedJobs []string `json:"targetReplicatedJobs"`
+
+    // Override parameters for Containers.
+    Containers []Container `json:"container,omitempty"`
+
+    // Override parameters for InitContainers.
+    InitContainer []Container `json:"initContainer,omitempty"`
+
+    // Override parameters for volumes.
+    Volumes []corev1.Volume `json:"volume,omitempty"`
+
+    // Custom Service Account
+    ServiceAccountName string `json:"serviceAccountName,omitempty"`
+}
+
+// Override for each container.
+// Parameters from Trainer, DatasetConfig, and ModelConfig will take precedence.
+type Container struct {
+
+    // Name for the container.
+    Name string `json:"name"`
+
+    // Command for the container.
+    Command []string `json:"command,omitempty" protobuf:"bytes,3,rep,name=command"`
+
+    // Args for the container.
+    Args []string `json:"args,omitempty"`
+
+    // Env for the container.
+    Env []corev1.EnvVar `json:"env,omitempty"`
+
+    // Env for the container.
+    EnvFrom []corev1.EnvFromSource `json:"envFrom,omitempty"`
+
+    // Override parameters for volume mounts.
+    VolumeMounts []VolumeMount `json:"volumeMounts,omitempty"`
+}
+```
+
+#### Example of TrainJob with Overrides
+
+This example shows how to override the user-identity for the sidecar container and add volume to the
+trainer container.
+
+```yaml
+apiVersion: kubeflow.org/v2alpha1
+kind: TrainJob
+metadata:
+  name: pytorch-distributed
+  namespace: tenant-alpha
+spec:
+  trainingRuntimeRef:
+    name: pytorch-distributed-gpu
+  trainer:
+    image: docker.io/custom-training
+  podSpecOverrides:
+    - targetReplicatedJobs:
+        - initializer
+          node
```

Comment:
```suggestion
```

Reply:
I think, we should remove initializer from here.